persistence/src/main/java/de/cyface/persistence/Persistence.java

```diff
@@ -0,0 +1,682 @@
+
+        final File[] finishedMeasurementDirs = synchronizedMeasurementsDir.listFiles(FileUtils.directoryFilter());
+
+        for (File measurement : finishedMeasurementDirs) {
+            final long measurementId = Long.parseLong(measurement.getName());
+            measurements.add(new Measurement(measurementId));
+        }
+
+        return measurements;
+    }
+
+    /**
+     * @return The content provider URI for the {@link IdentifierTable}
+     */
+    private static Uri getIdentifierUri(final String authority) {
+        return new Uri.Builder().scheme("content").authority(authority).appendPath(IdentifierTable.URI_PATH).build();
+    }
+
+    /**
+     * Loads a measurement with the provided id from the persistence layer as uncompressed and serialized in
+     * the {@link MeasurementSerializer#TRANSFER_FILE_FORMAT_VERSION} format, ready to be compressed.
+     *
+     * @param measurementId The identifier of the measurement to load.
+     * @return The bytes of the measurement in the <code>TRANSFER_FILE_FORMAT_VERSION</code> format.
+     * @throws FileCorruptedException If the persisted measurement if broken or in a false format.
+     * @throws IOException If the byte array could not be assembled.
+     */
+    public byte[] loadSerialized(final long measurementId) throws FileCorruptedException, IOException {
+        final byte[] transferFileFormat = new byte[2];
+        transferFileFormat[0] = (byte)(TRANSFER_FILE_FORMAT_VERSION >> 8);
+        transferFileFormat[1] = (byte)TRANSFER_FILE_FORMAT_VERSION;
+        final MetaFile.MetaData metaData = MetaFile.deserialize(context, measurementId);
+        final byte[] pointCounts = serialize(metaData.getPointMetaData());
+        final byte[] geoLocationData = metaData.getPointMetaData().getCountOfGeoLocations() > 0
+                ? FileUtils.loadBytes(GeoLocationsFile.loadFile(context, measurementId))
+                : new byte[] {};
+        final byte[] accelerationData = metaData.getPointMetaData().getCountOfAccelerations() > 0
+                ? FileUtils.loadBytes(AccelerationsFile.loadFile(context, measurementId))
+                : new byte[] {};
+        final byte[] rotationData = metaData.getPointMetaData().getCountOfRotations() > 0
+                ? FileUtils.loadBytes(RotationsFile.loadFile(context, measurementId))
+                : new byte[] {};
+        final byte[] directionData = metaData.getPointMetaData().getCountOfDirections() > 0
+                ? FileUtils.loadBytes(DirectionsFile.loadFile(context, measurementId))
+                : new byte[] {};
+
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        outputStream.write(transferFileFormat);
+        outputStream.write(pointCounts);
+        outputStream.write(geoLocationData);
+        outputStream.write(accelerationData);
+        outputStream.write(rotationData);
+        outputStream.write(directionData);
+        return outputStream.toByteArray();
```

Comment:
Hopefully this does not crash for large measurements.

Reply:
I added a TODO to test this on weak devices - all FIXMEs must be removed before release and all TODOs are checked and annotated with a Task before final release.